scripts/unpack_ddf.py

```diff
@@ -48,16 +53,125 @@
 BUILDERS['.materialc']      = render.material_ddf_pb2.MaterialDesc
 
 
-def print_dmanifest(manifest_file):
-    for field, data in manifest_file.ListFields():
-        if field.name == 'data':
-            t = resource.liveupdate_ddf_pb2.ManifestData()
-            t.MergeFromString(data)
-            print(field.name, ":")
-            print(t)
-
+proto_type_to_string_map = {}
+proto_type_to_string_map[google.protobuf.descriptor.FieldDescriptor.TYPE_BYTES]   = 'TYPE_BYTES'
+proto_type_to_string_map[google.protobuf.descriptor.FieldDescriptor.TYPE_DOUBLE]  = 'TYPE_DOUBLE'
+proto_type_to_string_map[google.protobuf.descriptor.FieldDescriptor.TYPE_ENUM]    = 'TYPE_ENUM'
+proto_type_to_string_map[google.protobuf.descriptor.FieldDescriptor.TYPE_FIXED32] = 'TYPE_FIXED32'
+proto_type_to_string_map[google.protobuf.descriptor.FieldDescriptor.TYPE_FIXED64] = 'TYPE_FIXED64'
+proto_type_to_string_map[google.protobuf.descriptor.FieldDescriptor.TYPE_FLOAT]   = 'TYPE_FLOAT'
+proto_type_to_string_map[google.protobuf.descriptor.FieldDescriptor.TYPE_GROUP]   = 'TYPE_GROUP'
+proto_type_to_string_map[google.protobuf.descriptor.FieldDescriptor.TYPE_INT32]   = 'TYPE_INT32'
+proto_type_to_string_map[google.protobuf.descriptor.FieldDescriptor.TYPE_INT64]   = 'TYPE_INT64'
+proto_type_to_string_map[google.protobuf.descriptor.FieldDescriptor.TYPE_MESSAGE] = 'TYPE_MESSAGE'
+proto_type_to_string_map[google.protobuf.descriptor.FieldDescriptor.TYPE_SFIXED32]= 'TYPE_SFIXED32'
+proto_type_to_string_map[google.protobuf.descriptor.FieldDescriptor.TYPE_SFIXED64]= 'TYPE_SFIXED64'
+proto_type_to_string_map[google.protobuf.descriptor.FieldDescriptor.TYPE_SINT32]  = 'TYPE_SINT32'
+proto_type_to_string_map[google.protobuf.descriptor.FieldDescriptor.TYPE_SINT64]  = 'TYPE_SINT64'
+proto_type_to_string_map[google.protobuf.descriptor.FieldDescriptor.TYPE_STRING]  = 'TYPE_STRING'
+proto_type_to_string_map[google.protobuf.descriptor.FieldDescriptor.TYPE_UINT32]  = 'TYPE_UINT32'
+proto_type_to_string_map[google.protobuf.descriptor.FieldDescriptor.TYPE_UINT64]  = 'TYPE_UINT64'
+
+TYPE_CONVERTERS = {}
+TYPE_CONVERTERS["dmLiveUpdateDDF.ManifestFile.data"] = resource.liveupdate_ddf_pb2.ManifestData
```

Comment:
This is the new printer setup. If it detects a message of type name `dmLiveUpdateDDF.ManifestFile.data`, then it will create a message of type `resource.liveupdate_ddf_pb2.ManifestData` and continue parsing.